### PR TITLE
Document core function options

### LINKED ISSUE #69: Document Core function options
Document the options for every high-level module function (if there are any)

### PR DESCRIPTION
**Pull request by cheezus1**
Decided to keep it simpler, by just adding types for the options. Parameter names and types are self-explanatory well enough, and easy to be accessed within the docs as well

resolves #69

### CODE REVIEW

**Comment by cheezus1** [comment]
> What about oracles? I can't see any changes.

I don't know how I've missed them :D

File: apps/core/lib/account.ex [hunk 1]
```diff
@@ -17,6 +17,11 @@ defmodule Core.Account do
   @allowed_recipient_tags ["ak", "ct", "ok", "nm"]
 
   @type account :: %{id: String.t(), balance: non_neg_integer(), nonce: non_neg_integer()}
+  @type spend_options :: [
+          fee: non_neg_integer(),
+          ttl: non_neg_integer(),
+          payload: non_neg_integer()
```
Comment (DanielaIvanova) [inline apps/core/lib/account.ex]: `payload` should be `String.t()`

File: apps/core/lib/oracle.ex [hunk 2]
```diff
@@ -33,6 +33,12 @@ defmodule Core.Oracle do
   @type sophia_data :: any()
   @type ttl_type :: :relative | :absolute
   @type ttl :: %{type: ttl_type(), value: non_neg_integer()}
+  @type oracle_options() :: [fee: non_neg_integer(), ttl: non_neg_integer()]
```
Comment (DanielaIvanova) [inline apps/core/lib/oracle.ex]: `ttl` should be `ttl()`

File: apps/core/lib/oracle.ex [hunk 3]
```diff
@@ -33,6 +33,12 @@ defmodule Core.Oracle do
...
+  @type query_options() :: [
+          fee: non_neg_integer(),
+          ttl: non_neg_integer(),
```
Comment (DanielaIvanova) [inline apps/core/lib/oracle.ex]: same as above

File: apps/core/lib/oracle.ex (hunk 2 above)
Comment (cheezus1) [reply apps/core/lib/oracle.ex]: yes, transaction `ttl`

File: apps/core/lib/oracle.ex (hunk 3 above)
Comment (cheezus1) [reply apps/core/lib/oracle.ex]: transaction `ttl`

**Comment by thepiwo** [comment]
will this be represented properly in autogenerated docs?

**Comment by cheezus1** [comment]
> will this be represented properly in autogenerated docs?

yes, types in function specs are clickable, redirecting you to the type definition

**Pull request merged**